src/core/featurelistextentcontroller.cpp

```diff
@@ -62,6 +62,29 @@ void FeatureListExtentController::zoomToSelected( bool skipIfIntersects ) const
   }
 }
 
+QgsPoint FeatureListExtentController::getCentroidFromSelected() const
+{
+  QgsPoint point( 0.0, 0.0, 0.0 );
+  if ( mModel && mSelection && mSelection->focusedItem() > -1 && mMapSettings )
+  {
+    QgsFeature feat = mSelection->focusedFeature();
+    QgsVectorLayer *layer = mSelection->focusedLayer();
+
+    if ( layer && layer->geometryType() != Qgis::GeometryType::Unknown && layer->geometryType() != Qgis::GeometryType::Null )
+    {
+      QgsGeometry geom = feat.geometry();
+
+      QgsCoordinateTransform transf( layer->crs(), mMapSettings->destinationCrs(), mMapSettings->mapSettings().transformContext() );
+      geom.transform( transf );
```

Comment:
Let's add some const.
```suggestion
    const QgsFeature feat = mSelection->focusedFeature();
    const QgsVectorLayer *layer = mSelection->focusedLayer();

    if ( layer && layer->geometryType() != Qgis::GeometryType::Unknown && layer->geometryType() != Qgis::GeometryType::Null )
    {
      QgsGeometry geom = feat.geometry();

      const QgsCoordinateTransform transf( layer->crs(), mMapSettings->destinationCrs(), mMapSettings->mapSettings().transformContext() );
      geom.transform( transf );
```